freshmaker/image.py

```diff
@@ -1554,6 +1564,66 @@ def _filter_out_already_fixed_published_images(
             child_image["parent"] = fixed_published_image
             del image_group[not_directly_affected_index:]
 
+    def _replace_base_images(self, to_rebuild, rpm_nvrs):
+        """
+        Replace base images that are not the latest and are used as dependency images.
+
+        :param Iterable to_rebuild: the list of images to rebuild; each element is
+            an iterable with the first element being the child image and each subsequent
+            image being the parent of the previous image
+        :param Iterable rpm_nvrs: the list of RPM NVRs with the fixes in the advisory
+        """
+        rpm_name_to_nvrs = {kobo.rpmlib.parse_nvr(nvr)["name"]: nvr for nvr in rpm_nvrs}
+
+        replacements = {}
+        for image_group in to_rebuild:
+            # Base image can only be present as the last image in list
+            image = image_group[-1]
+
+            if image.nvr in replacements:
+                new_image = replacements[image.nvr]
+                if not new_image:
+                    continue
+                image_group[-1] = new_image
+                image_group[-2]["parent"] = new_image
+                continue
+
+            # Skip non-base images
+            if not image.is_base_image:
```

Comment:
Aren't you sure already that this is a base image? Because it is in `image_group[-1]`.

If not, wouldn't it be better to move this if in the beginning of the for loop?

Reply:
Base image is always the last image in `image_group`, but it doesn't mean the last image is always base image, when the vulnerable rpms are not from base image, the last image is a "layered" image.

Reply:
Moved to the beginning of loop.